src/lib/forms/getFormTemplate.ts

```diff
@@ -0,0 +1,32 @@
+import { DatabaseConnectorClient } from "@lib/connectors/databaseConnector.js";
+import { logMessage } from "@src/lib/logger.js";
+import {
+  type FormTemplate,
+  formTemplateFromPostgreSqlResult,
+} from "./dataStructures/formTemplate.js";
+
+export async function getFormTemplate(
+  formId: string,
+): Promise<FormTemplate | undefined> {
+  return DatabaseConnectorClient.oneOrNone<Record<string, unknown>>(
+    'SELECT "jsonConfig" FROM "Template" WHERE id = $1',
+    [formId],
+  )
+    .then((result) => {
+      if (result === null) {
+        return undefined;
+      }
+
+      return formTemplateFromPostgreSqlResult(result);
+    })
+    .catch((error) => {
+      logMessage.error(
+        `[database] Failed to retrieve form template. FormId: ${formId}. Reason: ${JSON.stringify(
+          error,
+          Object.getOwnPropertyNames(error),
+        )}`,
+      );
+
+      throw error;
+    });
```

Comment:
Just curious why you went with `then/catch` here and if there's a preference on the team for this vs `await`.

Reply:
Following a recent discussion with @bryan-robitaille we decided to open the floor and start using the then/catch pattern when we feel like it would be appropriate. Through this PR, I am suggesting a place where I feel like it would be good to use it. If we agree then we will refactor our code to apply this in similar places.

Reply:
👨‍🍳 💋